File: frontend/ems-frontend/src/services/DepartmentService.js
import axios from "axios";

const DEPARTMENT_REST_API_BASE_URL = 'http://localhost:8080/api/departments'

export const listDepartments = () => axios.get(DEPARTMENT_REST_API_BASE_URL)

export const getDepartmentById = (departmentId) => axios.get(DEPARTMENT_REST_API_BASE_URL + '/' + departmentId)

export const createDepartment = (department) => axios.post(DEPARTMENT_REST_API_BASE_URL, department)

export const updateDepartment = (departmentId, department) => axios.put(DEPARTMENT_REST_API_BASE_URL + '/' + departmentId, department)

export const deleteDepartment = (departmentId) => axios.delete(DEPARTMENT_REST_API_BASE_URL + '/' + departmentId)

File: backend/src/main/java/com/springboot/emsbackend/service/DepartmentService.java
package com.springboot.emsbackend.service;

import com.springboot.emsbackend.dto.DepartmentDto;
import com.springboot.emsbackend.entity.Department;

import java.util.List;

public interface DepartmentService {
    DepartmentDto createDepartment(DepartmentDto departmentDto);

    DepartmentDto getDepartmentById(Long departmentId);

    List<DepartmentDto> getAllDepartments();

    DepartmentDto updateDepartment(Long departmentId, DepartmentDto updatedDepartment);

    void deleteDepartment(Long departmentId);
}
